### PR TITLE
Add 'noindex' meta tag to people pages that are less than 30 days old.

### LINKED ISSUE #4010: Add "noindex" meta tag to people and list pages belonging to accounts less than 30 days old
### Describe the problem that you'd like solved

Thousands of spam pages have been added to OL over time.  In many instances, people and list pages are where the spam shows up.  Spam can harm user trust in OL, affect OL's ranking in search engines, and potentially point to malicious sites and thus harm our users.

### Proposal & Constraints

In this week's community call, the decision was made to add the ["noindex" meta tag](https://en.wikipedia.org/wiki/Noindex) to people (and I think list) pages belonging to accounts less than 30 days old.  This will result in those pages not being indexed by search engines that honor that meta tag.

### Stakeholders

@mekarpeles @cdrini

### PR DESCRIPTION
**Pull request by SouthGoingZax**
Closes #4010

feature

### Technical

This adds the 'noindex' meta tag to newer people pages but not list pages.

### Testing

On dev, go to /recentchanges, click on a recently created people page, look at the source HTML, and search for "robots".  You should see:
```
    <meta name="robots" content="noindex" />
```
If testing in a local dev instance, see below.

### Screenshot

```
$ echo -- && grep days_since openlibrary/templates/type/user/view.html && curl -s localhost:8080/people/openlibrary | egrep "robots|Joined" && echo --
--
$if days_since(page.created) < 30:
    <i>Joined March 20, 2013</i>
--


$ echo -- && sed -i 's/< 30/< 2800/' openlibrary/templates/type/user/view.html && grep days_since openlibrary/templates/type/user/view.html && curl -s localhost:8080/people/openlibrary | egrep "robots|Joined" && echo --
--
$if days_since(page.created) < 2800:
    <meta name="robots" content="noindex" />
    <i>Joined March 20, 2013</i>
--

```

### Stakeholders

@cdrini @mekarpeles

### CODE REVIEW

**Review by mekarpeles — APPROVED** [review]
lgtm -- will add to dev + staging for testing prior to merge

File: openlibrary/core/helpers.py [hunk 1]
```diff
@@ -111,6 +111,12 @@ def safekey(x):
         return (k.__class__.__name__, k)
     return sorted(iterable, key=safekey, reverse=reverse)
 
+def days_since(then, now=None):
+    if now is None:
+        now = datetime.now()
+    delta = then - now
```
Comment (cclauss) [inline openlibrary/core/helpers.py:117]:
```suggestion
    delta = then - (now or datetime.now())
```

File: openlibrary/core/helpers.py [hunk 2]
```diff
@@ -111,6 +111,10 @@ def safekey(x):
         return (k.__class__.__name__, k)
     return sorted(iterable, key=safekey, reverse=reverse)
 
+def days_since(then, now=None):
+    delta = then - (now or datetime.now())
+    return abs(delta.days)
+
```
Comment (mekarpeles) [inline openlibrary/core/helpers.py:117]: Resolving:
```
git diff master -U0 | ./scripts/flake8-diff.sh
openlibrary/core/helpers.py:117:1: E302 expected 2 blank lines, found 1
1     E302 expected 2 blank lines, found 1
make: *** [Makefile:73: lint-diff] Error 1
```

```suggestion


```

File: openlibrary/core/helpers.py [hunk 3]
```diff
@@ -111,6 +111,11 @@ def safekey(x):
         return (k.__class__.__name__, k)
     return sorted(iterable, key=safekey, reverse=reverse)
 
+def days_since(then, now=None):
```
Comment (mekarpeles) [inline openlibrary/core/helpers.py:114]: Presumably extra line required above as well? Noting still a lint issue.

```suggestion

def days_since(then, now=None):
```

**Pull request merged**